knative-operator/pkg/controller/knativeeventing/knativeeventing_controller.go

```diff
@@ -26,7 +27,14 @@ import (
 // This needs to remain "knative-eventing-openshift" to be compatible with earlier versions.
 const finalizerName = "knative-eventing-openshift"
 
-var log = common.Log.WithName("controller")
+var (
+	log           = common.Log.WithName("controller")
+	stopTelemetry = make(chan struct{})
+	// Protects from processing order, if true we should install telemetry
+	// if it is false we need to uninstall it in the next delete stage.
+	// We start by assuming no telemetry is available.
+	shouldInstall = true
```

Comment:
I don't quite parse the need for this. Again this would be easier if we have a `Telemetry` type encapsulating everything and potentially keeping internal state.

Reply:
If you get a delete event first you need to avoid deleting stuff. If there is no chance I can remove it.

Reply:
Sure, but it feels quite odd to carry that state in the reconciler itself, seeing as it belongs to the telemetry controller in question.